tests/python_package_test/test_engine.py

```diff
@@ -1545,6 +1545,34 @@ def test_refit():
     assert err_pred > new_err_pred
 
 
+def test_refit_dataset_params():
+    # check refit accepts dataset_params
+    X, y = load_breast_cancer(return_X_y=True)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
+    lgb_train = lgb.Dataset(X_train, y_train)
+    params = {
+        'objective': 'binary',
+        'metric': 'binary_logloss',
+        'verbose': -1,
+        'min_data': 10
+    }
+    gbm = lgb.train(params, lgb_train, num_boost_round=20)
+    non_weight_err_pred = log_loss(y_test, gbm.predict(X_test))
+    new_gbm = gbm.refit(
+        data=X_train,
+        label=y_train,
+        weight=np.abs(np.random.normal(size=y_train.shape)),
```

Comment:
It seems that weights can be negative with `np.random.normal`.
`np.random.rand` should be a better option.
https://numpy.org/doc/stable/reference/random/generated/numpy.random.rand.html

Reply:
We don't want negative weights because it many in many cases the `hessian` value of a data is supposed to be non-negative. And `hessian` is multiplied by `weight`.

Reply:
@shiyu1994 ok, an additional explanation to `weight` parameter documentation could be added

```
weight : list, numpy 1-D array, pandas Series or None, optional (default=None)
            Weight for each ``data`` instance. Weight should be non-negative values because the Hessian
            is supposed to be non-negative.
```